nixos/modules/services/web-apps/librechat.nix

```diff
@@ -0,0 +1,232 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librechat;
+  format = pkgs.formats.yaml { };
+  configFile = format.generate "librechat.yaml" cfg.settings;
+  exportCredentials = n: _: ''export ${n}="$(${pkgs.systemd}/bin/systemd-creds cat ${n}_FILE)"'';
+  exportAllCredentials = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList exportCredentials vars);
+  getLoadCredentialList = lib.mapAttrsToList (n: v: "${n}_FILE:${v}") cfg.credentials;
+in
+{
+  options.services.librechat = {
+    enable = lib.mkEnableOption "the LibreChat server";
+
+    package = lib.mkPackageOption pkgs "librechat" { };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Whether to open the port in the firewall.
+      '';
+    };
```

Comment:
Nit:


```suggestion
    openFirewall = lib.mkEnableOption "firewall rules for LibreChat";
```

Reply:
It  was like this initialy, but @pluiedev suggested a better message in https://github.com/NixOS/nixpkgs/pull/412668#discussion_r2120830912 and I didn't like that "hacky syntax" and suggested to make it a normal option in https://github.com/NixOS/nixpkgs/pull/412668#discussion_r2462977699.